test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Feeds/InstalledAndTransitivePackageFeedTests.cs

```diff
@@ -171,5 +170,69 @@ public async Task SearchAsync_WithTransitiveData_SucceedsAsync()
                 elem => Assert.IsType<ClonedPackageSearchMetadata>(elem),
                 elem => Assert.IsType<TransitivePackageSearchMetadata>(elem));
         }
+
+        [Fact]
+        public async Task GetPackageMetadataAsync_WithCancellationToken_ThrowsAsync()
+        {
+            var testPackageIdentity = new PackageCollectionItem("FakePackage", new NuGetVersion("1.0.0"), null);
+            var testTransitiveIdentity = new PackageCollectionItem("TransitiveFakePackage", new NuGetVersion("1.0.0"), null);
+            var _target = new InstalledAndTransitivePackageFeed(new[] { testPackageIdentity }, new[] { testTransitiveIdentity }, _packageMetadataProvider);
+
+            using CancellationTokenSource cts = new CancellationTokenSource();
+            cts.Cancel();
+            await Assert.ThrowsAsync<OperationCanceledException>(async () => await _target.GetPackageMetadataAsync(It.IsAny<PackageIdentity>(), It.IsAny<bool>(), cts.Token));
+            await Assert.ThrowsAsync<OperationCanceledException>(async () => await _target.GetPackageMetadataAsync(It.IsAny<PackageCollectionItem>(), It.IsAny<bool>(), cts.Token));
+        }
+
+        [Theory]
+        [InlineData(new string[] {}, new[] {"pkg1", "pkg2", "pkg3"}, "", 0, 3)]
+        [InlineData(new[] {"Gamma", "Beta", "Alfa", "Delta"}, new string[] {}, "", 4, 0)]
+        [InlineData(new[] {"Gamma", "Beta", "Alfa", "Delta"}, new[] {"pkg1", "pkg2", "pkg3"}, "", 4, 3)]
+        [InlineData(new[] {"Gamma", "Beta", "Alfa", "Delta"}, new[] {"pkg1", "pkg2", "pkg3"}, "g", 1, 3)]
+        [InlineData(new[] {"Gamma", "Beta", "Alfa", "Delta"}, new[] {"pkg1", "pkg2", "pkg3"}, "#@", 0, 0)]
+        [InlineData(new[] {"Gamma", "Beta", "Alfa", "Delta"}, new[] {"pkg1", "pkg2", "pkg3"}, "alFA", 1, 0)]
+        [InlineData(new[] {"Gamma", "Beta", "Alfa", "Delta"}, new[] {"pkg1", "pkg2", "pkg3"}, "pkg1", 0, 1)]
+        [InlineData(new[] {"Beta", "Alfa", "Delta", "Gamma"}, new[] {"pkg2", "pkg3", "pkg1"}, "ta", 2, 0)]
+        [InlineData(new[] {"Beta", "Alfa", "Delta", "Gamma"}, new[] {"pkg3", "pkg2", "pkg1"}, "g", 1, 3)]
+        [InlineData(new[] {"q", "z", "hi"}, new[] { "t" }, "z", 1, 0)]
```

Comment:
Can we add another InlineData (or a few if necessary) for installed / transitive packages such that the packages are alphabetically interleaving? That way, we can ensure that if a bug where introduced where the search results were returned by alphabetical order, the test would catch that.